fuel-vm/src/tests/debugger.rs

```diff
@@ -0,0 +1,72 @@
+use alloc::{
+    vec,
+    vec::Vec,
+};
+
+use fuel_asm::{
+    op,
+    RegId,
+};
+use fuel_tx::{
+    ConsensusParameters,
+    Finalizable,
+    GasCosts,
+    Script,
+    TransactionBuilder,
+};
+
+use crate::{
+    prelude::{
+        Interpreter,
+        IntoChecked,
+    },
+    state::ProgramState,
+};
+
+#[test]
+fn receipts_are_produced_correctly_with_stepping() {
+    let script = vec![
+        op::movi(0x20, 1234),
+        op::log(0x20, RegId::ZERO, RegId::ZERO, RegId::ZERO),
```

Comment:
Can we add more than one log to be sure it's ordered even more than just : "all before the return one"

Reply:
This test case is enought to demonstrate the bug. The issue was that the script result receipt would be created right after the first call to `transact` returns, i.e. before the log.